dz13/src/app.js

```diff
@@ -0,0 +1,121 @@
+function isSomeProductInStock(productsList) {
+  return productsList.some((product) => product.inStock)
+}
+
+function makeUserTotalBalance(transactions) {
+  const newTransactions = transactions.reduce((result, transaction) => {
+    if (transaction.type === 'income') {
+      return result + transaction.amount
+    }
+    return result - transaction.amount
+  }, 0)
+  return newTransactions
+}
+
+function entriesArray(array) {
+  return array.flat(Infinity)
+}
+function getObjectKeysOrValues(object, type) {
+  return type === 'key' ? Object.keys(object) : Object.values(object)
+}
+
+function logInCheck(carts) {
+  Object.entries(carts).forEach((cart) => (console.log(`${cart[0]}: ${cart[1]} шт.`)))
+}
```

Comment:
```suggestion
function logCartProducts(cart) {
  Object.entries(cart).forEach(([product, count]) => {
    console.log(`${product}: ${count} шт.`)
  })
}
```

заменил название функции - logIn обычно подразумевает вход

убрал return в forEach - он не возвращает, просто перебирает

применил деструктуризацию массива (элемента списка продуктов) в формате [product, count] - получил наглядное название переменных вместо cart[0], cart[1]